pkg/resource/api.go

```diff
@@ -102,6 +106,58 @@ func (lo NamespacePatcher) Patch(cr ParentResource, list []ChildResource) ([]Chi
 	return list, nil
 }
 
+// NewLabelPropagator returns a new LabelPropagator
+func NewLabelPropagator() LabelPropagator {
+	return LabelPropagator{}
+}
+
+// LabelPropagator propagates all the labels that the parent resource has down
+// to all child resources.
+type LabelPropagator struct{}
+
+func (lo LabelPropagator) Patch(cr ParentResource, list []ChildResource) ([]ChildResource, error) {
+	for _, o := range list {
+		labels := o.GetLabels()
+		if labels == nil {
+			labels = map[string]string{}
+		}
+		for key, val := range cr.GetLabels() {
+			labels[key] = val
+		}
+		o.SetLabels(labels)
+	}
+	return list, nil
+}
+
+// NewParentLabelSetAdder returns a new ParentLabelSetAdder
+func NewParentLabelSetAdder() ParentLabelSetAdder {
+	return ParentLabelSetAdder{}
+}
+
+// ParentLabelSetAdder adds parent labels to the child resources.
+// See https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-stack-relationship-labels.md
+type ParentLabelSetAdder struct{}
+
+func (lo ParentLabelSetAdder) Patch(cr ParentResource, list []ChildResource) ([]ChildResource, error) {
+	for _, o := range list {
+		labels := o.GetLabels()
+		if labels == nil {
+			labels = map[string]string{}
+		}
+		if cr.GetNamespace() != "" {
+			labels[stacks.LabelParentNamespace] = cr.GetNamespace()
+		}
+		labels[stacks.LabelParentName] = cr.GetName()
+		labels[stacks.LabelParentUID] = string(cr.GetUID())
+		gvk := cr.GetObjectKind().GroupVersionKind()
+		labels[stacks.LabelParentKind] = gvk.Kind
+		labels[stacks.LabelParentGroup] = gvk.Group
+		labels[stacks.LabelParentVersion] = gvk.Version
+		o.SetLabels(labels)
```

Comment:
`stack.ParentLabels` (crossplane) and `meta.AddLabels` (crossplane-runtime) could minimize the code needed here.

Reply:
`meta.AddLabels(o, stacks.ParentLabels(cr))`